### PR TITLE
chore: Refactor examples to leverage `terraform-aws-eks` for cluster creation

### LINKED ISSUE #1076: Cannot Create Log Stream due to Double Quote
https://github.com/aws-ia/terraform-aws-eks-blueprints/blob/0848d87e4ab52f76dfff2e3d0c6d64d196b6e877/examples/complete-kubernetes-addons/main.tf#L189

The double quotation marks in the log_stream_prefix field in the example of the Fargate Fluentbit addon causing the SerializationException during the log stream creation. It works fine if the double quotation marks are removed.

### PR DESCRIPTION
**Pull request by bryantbiggs**
### What does this PR do?

- Refactor examples to leverage `terraform-aws-eks` for cluster creation
- Add recommended security group rules for node groups - this is planned to be available as a default in v19 of `terraform-aws-modules`
	
### Motivation

- Relates to #868
- Updating examples to start showing what the future plan is for examples/blueprints without introducing any breaking changes to users 
- Resolves #1076

### More

- [ ] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [ ] Yes, I have added a new example under [examples](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/examples) to support my PR
- [ ] Yes, I have created another PR for add-ons under [add-ons](https://github.com/aws-samples/eks-blueprints-add-ons) repo (if applicable)
- [ ] Yes, I have updated the [docs](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/docs) for this feature
- [x] Yes, I ran `pre-commit run -a` with this PR

**Note**: Not all the PRs require a new example and/or doc page. In general:
- Use an existing example when possible to demonstrate a new addons usage
- A new docs page under `docs/add-ons/*` is required for new a new addon

### For Moderators

- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -99,8 +40,6 @@ EKS Blueprints for Terraform is maintained by AWS Solution Architects. It is not
 
 To post feedback, submit feature ideas, or report bugs, please use the [Issues section](https://github.com/aws-ia/terraform-aws-eks-blueprints/issues) of this GitHub repo.
 
-For architectural details, step-by-step instructions, and customization options, see our [documentation site](https://aws-ia.github.io/terraform-aws-eks-blueprints/).
```
Comment (bryantbiggs) [inline README.md:102]: duplicate statement

File: examples/agones-game-controller/main.tf [hunk 2]
```diff
@@ -85,15 +70,26 @@ module "eks_blueprints_kubernetes_addons" {
...
-
   # Add-ons
-  enable_metrics_server         = true
-  enable_cluster_autoscaler     = true
-  enable_aws_cloudwatch_metrics = true
+  enable_metrics_server     = true
+  enable_cluster_autoscaler = true
+
+  # NOTE: Agones requires a Node group in Public Subnets and enable Public IP
+  enable_agones = true
```
Comment (bryantbiggs) [inline examples/agones-game-controller/main.tf:78]: I don't know why GitHub does this with diffs - the `upstream-with-k8s-addons` was deleted, not renamed

File: examples/complete-kubernetes-addons/main.tf [hunk 3]
```diff
@@ -178,40 +111,6 @@ module "eks_blueprints_kubernetes_addons" {
     ]
   }
 
-  enable_fargate_fluentbit = true
```
Comment (bryantbiggs) [inline examples/complete-kubernetes-addons/main.tf:181]: moved to `fargate-serverless`

**Comment by bryantbiggs** [comment]
failed plans will fail since those are removed or re-named

**Review by Zvikan — COMMENTED** [review]
This is a big one! LGTM, some minor comments and questions here.
Let's make sure to run the e2e across the examples and do some sanity checks to all the running resources per example, feel free to pull me in so we can divide and conquer this.

File: .github/workflows/e2e-parallel-destroy.yml [hunk 4]
```diff
@@ -29,16 +29,11 @@ jobs:
           - example_path: examples/eks-cluster-with-new-vpc
           - example_path: examples/fargate-serverless
           # - example_path: examples/fully-private-eks-cluster # skipping until issue #711 is addressed
-          - example_path: examples/game-tech/agones-game-controller
           - example_path: examples/gitops/argocd
           # - example_path: examples/ingress-controllers/nginx # ignoring due to https://github.com/kubernetes-sigs/aws-load-balancer-controller/issues/1629
```
[the remark this answers is not in the document]
Comment (bryantbiggs) [reply .github/workflows/e2e-parallel-destroy.yml]: fixed in https://github.com/aws-ia/terraform-aws-eks-blueprints/pull/1074/commits/ffec553ae703e3ed853e63c29518b705ee9b7a22

File: .github/workflows/e2e-parallel-full.yml [hunk 5]
```diff
@@ -33,16 +33,11 @@ jobs:
           - example_path: examples/eks-cluster-with-new-vpc
           - example_path: examples/fargate-serverless
           # - example_path: examples/fully-private-eks-cluster # skipping until issue #711
-          - example_path: examples/game-tech/agones-game-controller
           - example_path: examples/gitops/argocd
           # - example_path: examples/ingress-controllers/nginx # ignoring due to https://github.com/kubernetes-sigs/aws-load-balancer-controller/issues/1629
```
[the remark this answers is not in the document]
Comment (bryantbiggs) [reply .github/workflows/e2e-parallel-full.yml]: fixed in https://github.com/aws-ia/terraform-aws-eks-blueprints/pull/1074/commits/ffec553ae703e3ed853e63c29518b705ee9b7a22

**Comment by bryantbiggs** [comment]
switching to draft to split up PR

**Pull request closed without merging**